src/python/bot/fuzzers/libfuzzer.py

```diff
@@ -321,6 +323,73 @@ def fuzz(self,
                                 artifact_prefix, additional_args, extra_env)
 
 
+class FuchsiaQemuLibFuzzerRunner(new_process.ProcessRunner, LibFuzzerCommon):
+  """libFuzzer runner (when Fuchsia is the target platform)."""
+
+  def __init__(self, executable_path, default_args=None):
+    if (not environment.get_value('FUCHSIA_PKEY_PATH')) or (
+        not environment.get_value('FUCHSIA_PORTNUM')):
+      raise fuchsia.errors.FuchsiaConfigError(
+          'FUCHSIA_PKEY_PATH and/or FUCHSIA_PORTNUM was not set')
+    # yapf: disable
+    self.ssh_root = [
+        'ssh',
+        '-i', environment.get_value('FUCHSIA_PKEY_PATH'),
+        '-o', 'StrictHostKeyChecking no',
+        '-o', 'UserKnownHostsFile=/dev/null',
+        '-p', str(environment.get_value('FUCHSIA_PORTNUM')),
+        'localhost'
+    ]
+    # yapf: enable
+
+    super(FuchsiaQemuLibFuzzerRunner, self).__init__(
+        executable_path=executable_path, default_args=default_args)
+
+  def get_command(self, additional_args=None):
+    # TODO(flowerhack): Update this to dynamically pick a result from "fuzz
+    # list" and then run that fuzzer.
+    return self.ssh_command('ls')
+
+  def fuzz(self,
+           corpus_directories,
+           fuzz_timeout,
+           artifact_prefix=None,
+           additional_args=None,
+           extra_env=None):
+    """LibFuzzerCommon.fuzz override."""
+
+    # Don't try to run the fuzzer until we know we can SSH successfully.
+    ssh_test_process = new_process.ProcessRunner('ssh',
+                                                 self.ssh_command('ls')[1:])
+    # Number of retries chosen somewhat arbitrarily; if we get too many
+    # spurious exceptions, we can tweak this.
+    final_result_return_code = ''
+    for _ in range(30):
+      result = ssh_test_process.run_and_wait()
+      if (not result.return_code) and (not result.timed_out):
+        # SSH works; start fuzzing.
+        return LibFuzzerCommon.fuzz(self, corpus_directories, fuzz_timeout,
+                                    artifact_prefix, additional_args)
+      else:
+        final_result_return_code = result.return_code
+        time.sleep(2)
```

Comment:
use @retry.wrap() for the retry logic. 

example: https://github.com/google/clusterfuzz/blob/master/src/python/google_cloud_utils/credentials.py